File: java/chaincue-real-estate/src/main/java/com/chaincuerealestate/chaincuerealestate/repositories/CountryRepository.java
package com.chaincuerealestate.chaincuerealestate.repositories;

import com.chaincuerealestate.chaincuerealestate.domains.Country;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface CountryRepository extends JpaRepository<Country, String> {
    Optional<Country> findByName(String name);
}

File: kotlin/chaincue-real-estate/src/main/kotlin/com/example/chaincuerealestate/repositories/CountryRepository.kt
package com.example.chaincuerealestate.repositories

import com.example.chaincuerealestate.domains.Country
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface CountryRepository : JpaRepository<Country, String> {
    fun findByName(name: String): Optional<Country>
}

